openedx/frontend-app-library-authoring
# feat: add AbortController to library load apis

JIRA Ticket: [TNL-11082](https://2u-internal.atlassian.net/browse/TNL-11082)

This PR add the ability to cancel a previous libraries api that have not complete before the load of the current library. The cancels only apply to api calls that run on the initial page load.

Testing

1. Using the course import function, add numerous courses to your current library.
2. While waiting for all of those courses to import, make sure that you have at least one other library created. IF not, create a library.
3. Go back to your first library and check that the courses finished importing and check that there are multiple pages in the available in library authoring page.
4. Open the browser developer tools.
5. Once there are many pages in the library authoring page, go back to the library select page.
6. Choose your super large library.
7. While the page is loading, navigate back to the library select page and choose a second course.
8. Allow the second course to load.
9. Check the network tab of developer tools and confirm that the api calls for the first page were cancelled.
10. Check that page for the second library loaded as expected (visually).
